src/gnark_backend_glue/mod.rs

```diff
@@ -83,21 +85,30 @@ pub fn prove_with_pk(
     circuit: &Circuit,
     values: Vec<FieldElement>,
     proving_key: &[u8],
-) -> Result<Vec<u8>> {
-    let rawr1cs = RawR1CS::new(circuit.clone(), values)?;
+) -> Result<Vec<u8>, GnarkBackendError> {
+    let rawr1cs = RawR1CS::new(circuit.clone(), values).map_err(|_e| GnarkBackendError::Error)?;
```

Comment:
`RawR1CS::new` should return a `GnarkBackendError` instead of an `anyhow::Error`.